pyomo/contrib/fme/plugins/fourier_motzkin_elimination.py

```diff
@@ -0,0 +1,329 @@
+        cons_dict = {'lower': constraint.lower,
+                     'upper': constraint.upper,
+                     'body': std_repn
+                     }
+        constraints_to_add = [cons_dict]
+        if cons_dict['upper'] is not None:
+            # if it has both bounds
+            if cons_dict['lower'] is not None:
+                # copy the constraint and flip
+                leq_side = {'lower': -cons_dict['upper'],
+                            'upper': None,
+                            'body': generate_standard_repn(-1.0*body)}
+                self._move_constant_and_add_map(leq_side)
+                constraints_to_add.append(leq_side)
+                cons_dict['upper'] = None
+
+            # If it has only an upper bound, we just need to flip it
+            else:
+                # just flip the constraint
+                cons_dict['lower'] = -cons_dict['upper']
+                cons_dict['upper'] = None
+                cons_dict['body'] = generate_standard_repn(-1.0*body)
+        self._move_constant_and_add_map(cons_dict)
+
+        return constraints_to_add
+
+    def _move_constant_and_add_map(self, cons_dict):
+        """Takes constraint in dicionary form already in >= form, 
+        and moves the constant to the RHS
+        """
+        body = cons_dict['body']
+        constant = body.constant
+        cons_dict['lower'] -= constant
+        body.constant = 0
+
+        # store a map of vars to coefficients. We can't use this in place of
+        # standard repn because determinism, but this will save a lot of linear
+        # time searches later.
+        cons_dict['map'] = ComponentMap(zip(body.linear_vars, body.linear_coefs))
+
+    def _fourier_motzkin_elimination(self, constraints, vars_to_eliminate):
+        """Performs FME on the constraint list in the argument 
+        (which is assumed to be all >= constraints and stored in the 
+        dictionary representation), projecting out each of the variables in 
+        vars_to_eliminate"""
+
+        # We only need to eliminate variables that actually appear in
+        # this set of constraints... Revise our list.
+        vars_that_appear = []
+        for cons in constraints:
+            for var in cons['body'].linear_vars:
+                if var in vars_to_eliminate:
+                    vars_that_appear.append(var)
+
+        # we actually begin the recursion here
+        while vars_that_appear:
+            # first var we will project out
+            the_var = vars_that_appear.pop()
+
+            # we are 'reorganizing' the constraints, we will map the coefficient
+            # of the_var from that constraint and the rest of the expression and
+            # sorting based on whether we have the_var <= other stuff or vice
+            # versa.
+            leq_list = []
+            geq_list = []
+            waiting_list = []
+
+            while(constraints):
+                cons = constraints.pop()
```

Comment:
Why `pop`?  Couldn't you just iterate over the constraints and at the end of the loop replace the constraints with 
```python
constraints = waiting_list
for leq in leq_list:
    for geq in geq_list:
        constraints.append(self._add_linear_constraints(leq, geq))
```

Reply:
Ha! Oops, yes, that makes much more sense. :)